i18n(pt-BR): Translate `guides/content-collections.mdx` (#3998)
withastro/docs: src/content/docs/pt-br/guides/content-collections.mdx

```ts
// src/content/config.ts
const colecaoBlog = defineCollection({
  type: 'content',
  schema: z.object({ /* ... */ })
});
const jornal = defineCollection({
  type: 'content',
  schema: z.object({ /* ... */ })
});
const autores = defineCollection({
  type: 'data',
  schema: z.object({ /* ... */ })
});

export const collections = {
  'blog': colecaoBlog,
  'jornal': jornal,
  'autores': autores,
};
```

Conforme seu projeto cresce, você também é libre para reorganizar sua base de código e mover a lógica do arquivo `src/content/config.ts`. Definir seus esquemas separadamente pode ser útil para reutilizar esquemas através de múltiplas coleções e compartilhar esquemas com outras partes do seu projeto.

```ts
// src/content/config.ts
// 1. Importe seus utilitários e esquemas
import { defineCollection } from 'astro:content';
import { esquemaBlog, esquemaAutor } from '../schemas';

// 2. Defina suas coleções
const colecaoBlog = defineCollection({
  type: 'content',
  schema: esquemaBlog,
});
const colecaoAutor = defineCollection({
  type: 'data',
  schema: esquemaAutor,
});

> // 3. Export várias coleções para registrá-las

Reviewer: ```suggestion
// 3. Exporte várias coleções para registrá-las
```

```ts
import { defineCollection, reference, z } from 'astro:content';

const blog = defineCollection({
  type: 'content',
  schema: z.object({
    titulo: z.string(),
    // Referencia um único autor da coleção `autores` pelo `id`
    autor: reference('autores'),
    // Referencia um array de postagens relacionadas da coleção `blog` pelo `slug`
    postagensRelacionadas: z.array(reference('blog')),
  })
});

const autores = defineCollection({
  type: 'data',
  schema: z.object({
    nome: z.string(),
    portfolio: z.string().url(),
  })
});

export const collections = { blog, autores };
```

Este exemplo de postagem de blog especifica os `slug`s de postagens relacionadas e o `id` do autor da postagem:
This example blog post specifies the `slug`s of related posts and the `id` of the post author:

```yaml title="src/content/blog/bem-vindo.md"
---
titulo: "Bem-vindo ao meu blog"
autor: ben-holmes # referencia `src/content/autores/ben-holmes.json`
postagensRelacionadas:
- sobre-mim # referencia `src/content/blog/some-mim.md`

> - minha-retrospectiva # references `src/content/blog/minha-retrospectiva.md`

Reviewer: ```suggestion
- minha-retrospectiva # referencia `src/content/blog/minha-retrospectiva.md`
```

    - semana-2.md uma entrada da coleção
    - semana-3.md uma entrada da coleção
</FileTree>

Tendo a coleção, você pode começar a [consultar seu conteúdo](#consultando-coleções) usando as APIs de conteúdo integradas no Astro.

### A Pasta ".astro"

O Astro armazena metadados importantes para coleções de conteúdo em uma pasta `.astro` no seu projeto. Nenhuma ação é necessária da sua parte para manter ou atualizar essa pasta. Você é encorajado a ignorá-lo completamente enquanto estiver trabalhando no seu projeto.

A pasta `.astro` vai ser atualizada para você automaticamente sempre que você executa os comandos [`astro dev`](/pt-br/reference/cli-reference/#astro-dev) e [`astro build`](/pt-br/reference/cli-reference/#astro-build). Você pode executar [`astro sync`](/pt-br/reference/cli-reference/#astro-sync) a qualquer momento para atualizar a pasta `.astro` manualmente.

:::tip
Se você está usando Git para controle de versão, nós recomendamos ignorar a pasta `.astro` adicionando `.astro` no seu `.gitignore`. Isso diz ao Git ignorar essa pasta e quaisquer arquivos dentro dela.

```bash
echo "\n.astro" >> .gitignore
```
:::

### Organizando com múltiplas coleções

> Se dois arquivos representam diferentes tipos de conteúdo (e.x. uma postagem de blog e um perfil de autor), eles provavelmente pertencem a coleções diferentes. Isso é importante por que muitas funcionalidades (validação do frontmatter, segurança automática de tipo do TypeScript) requerem que todas as entradas em uma coleção compartilhem uma estrutura similar.

Reviewer: Porque (junto) é o correto para essa frase
```suggestion
Se dois arquivos representam diferentes tipos de conteúdo (e.x. uma postagem de blog e um perfil de autor), eles provavelmente pertencem a coleções diferentes. Isso é importante porque muitas funcionalidades (validação do frontmatter, segurança automática de tipo do TypeScript) requerem que todas as entradas em uma coleção compartilhem uma estrutura similar.
```

Author: 😡

Você pode sobrescrever a slug gerada para a entrada ao adicionar seu própria propriedade `slug` ao frontmatter do arquivo. Isso é similar à funcionalidade `permalink` de outros frameworks web. `"slug` é um nome de propriedade especial e reservada que não é permitida no seu `schema` customizado da coleção e não vai aparecer na propriedade `data` da sua entrada.

```md {3}
---
title: Minha Postagem de Blog
slug: minha-slug-customizada/suporta/barras
---
Seu conteúdo da postagem do blog vem aqui.
```

## Consultando Coleções

O Astro fornece duas funções para consultar uma coleção e retornar um (ou mais) entradas de conteúdo: [`getCollection()`](/pt-br/reference/api-reference/#getcollection) e [`getEntry()`](/pt-br/reference/api-reference/#getentry).

```js
import { getCollection, getEntry } from 'astro:content';

// Obtém todas as entradas de uma coleção.
// Requer o nome da coleção como um argumento.
// Exemplo: recuperar `src/content/blog/**`
const todasPostagensDoBlog = await getCollection('blog');

// Obtém uma única entrada de uma coleção.

> // Requires the name of the collection and either

Reviewer: Pedaço de inglês perdido
```suggestion

```

```astro title="src/pages/blog/bem-vindo.astro"
---
import { getEntry, getEntries } from 'astro:content';

const postagemBlog = await getEntry('blog', 'bem-vindo');

// Recupera um única referencia
const autor = await getEntry(postagemBlog.data.autor);
// Recupera um array de referências
const postagensRelacionadas = await getEntries(postagemBlog.data.postagensRelacionadas);
---

<h1>{postagemBlog.data.titulo}</h1>
<p>Autor: {autor.data.nome}</p>

<!-- ... -->

<h2>Você pode também gostar:</h2>
{postagensRelacionadas.map(p => (
  <a href={p.slug}>{p.data.titulo}</a>
))}
```

### Filtrando consultas de coleção

A `getCollection()` recebe um callback "filter" opcional que permite você filtrar sua consulta baseado em um `id` ou propriedades do `data` (frontmatter) da entrada. Para coleções `type: 'content'`, você também pode filtrar baseado no `slug`.

:::note
A propriedade `slug` é específica para coleções de conteúdo e não vai estar disponível ao filtrar coleções de JSON ou YAML.
:::

> Você pode usar isso para filtrar por qualquer critério de conteúdo que você quiser. Por exemplo, você pode filtrar por propriedades como `rascunho` para evitar que qualquer postagens do blog em rascunho sejam publicadas para seu blog:

Reviewer: A propriedade `draft` vem por padrão com o Astro, não é definida pelo usuário, então eu manteria a palavra em inglês, já que contextualizamos na mesma frase o que significa. 
https://docs.astro.build/en/guides/markdown-content/#draft-pages
```suggestion
Você pode usar isso para filtrar por qualquer critério de conteúdo que você quiser. Por exemplo, você pode filtrar por propriedades como `draft` para evitar que quaisquer postagens do blog em rascunho sejam publicadas para seu blog:
```

```js

> // Exemplo: Filtrar entradas de conteúdo com `rascunho: true`
> import { getCollection } from 'astro:content';
> const publishedBlogEntries = await getCollection('blog', ({ data }) => {
>   return data.rascunho !== true;
> });

Reviewer: ```suggestion
// Exemplo: Filtrar entradas de conteúdo com `draft: true`
import { getCollection } from 'astro:content';
const entradasPublicadasBlog = await getCollection('blog', ({ data }) => {
  return data.draft !== true;
});
```

  return id.startsWith('en/');
});
```

### Usando conteúdo em modelos do Astro

Uma vez que você tenha consultado suas entradas de coleção, você pode acessar cada entrada diretamente do template do seu componente Astro. Isso permite que você renderize HTML para coisas como links para seu conteúdo (usando o `slug` do conteúdo) ou informação sobre seu conteúdo (usando a propriedade `data`).

Para informações sobre renderizando seu conteúdo para HTML, veja [Renderizando Conteúdo para HTML](/pt-br/guides/content-collections/#renderizando-conteúdo-para-html) abaixo.

```astro
---
// src/pages/index.astro
import { getCollection } from 'astro:content';
const entradasBlog = await getCollection('blog');
---
<ul>
  {entradasBlog.map(entradaPostagemBlog => (
    <li>
      <a href={`/minha-url-do-blog/${entradaPostagemBlog.slug}`}>{entradaPostagemBlog.data.titulo}</a>
      <time datetime={entradaPostagemBlog.data.dataPublicacao.toISOString()}>
        {entradaPostagemBlog.data.dataPublicacao.toDateString()}
      </time>
    </li>
  ))}
</ul>
```

### Passando conteúdo como props

Um componente também pode passar uma entrada de conteúdo inteira como prop.

> Se você fizer isso, você pode usar o utilitário [`CollectionEntry`](/pt-br/reference/api-reference/#tipo-da-entrada-da-coleção) para definir o tipo corretamente das props dos seus componentes usando TypeScript. Esse utilitário recebe uma string como argumento que corresponde com o nome do esquema da sua coleção e vai herder todas as propriedades desse esquema de coleção.

Reviewer: ```suggestion
Se você fizer isso, você pode usar o utilitário [`CollectionEntry`](/pt-br/reference/api-reference/#tipo-da-entrada-da-coleção) para definir o tipo corretamente das props dos seus componentes usando TypeScript. Esse utilitário recebe uma string como argumento que corresponde com o nome do esquema da sua coleção e vai herdar todas as propriedades desse esquema de coleção.
```

```astro
---
// src/pages/postagens/[...slug].astro
import { getEntry } from "astro:content";
// 1. Pega o slug da requisição recebida pelo servidor
const { slug } = Astro.params;
if (slug === undefined) {
	throw new Error("O slug é obrigatório");
}
// 2. Consulta pela entrada diretamente usando o slug da requisição
const entrada = await getEntry("blog", slug);
// 3. Redireciona se a entrada não existir
if (entrada === undefined) {
	return Astro.redirect("/404");
}
// 4. (Opcional) Renderiza a entrada para HTML no template
const { Content } = await entrada.render();
---
```

## Migrando do Roteamento Baseado em Arquivos
Esse guia mostra como converter um projeto com Astro existente com arquivos Markdown na pasta `src/pages/` para coleções de conteúdo. Ele usa o [projeto final do tutorial Construa um Blog](https://github.com/withastro/blog-tutorial-demo) como exemplo.

> 1. [Atualize](/pt-br/guides/upgrade-to/v2/) para o Astro v2.0 ou posterior, e atualize todas as integrações para suas verões mais recente.

Reviewer: ```suggestion
1. [Atualize](/pt-br/guides/upgrade-to/v2/) para o Astro v2.0 ou posterior, e atualize todas as integrações para suas versões mais recentes.
```

2. [Configure o TypeScript](/pt-br/guides/content-collections/#configurando-o-typescript) para coleções de conteúdo.
3. Crie pelo menos uma coleção (pasta em `src/content/`) e mova suas páginas em Markdown e MDX da `src/pages/` para essas subpastas do `src/content/`. Coleções funcionam melhor quando todos os arquivos na mesma coleção possuem propriedades do frontmatter similares. Então, escolha sua nova estrutura de pastas para refletir tipos de páginas similares.

    Por exemplo, para migrar a [postagem de blog no tutorial](/pt-br/tutorial/2-pages/2/), mova o conteúdo de `src/pages/posts/` para `src/content/postagens/`.

4. Crie um arquivo `src/content/config.ts` e [defina ume esquema](/pt-br/guides/content-collections/#definindo-um-esquema-de-coleção) para cada tipo de conteúdo. Para o blog, nós temos um tipo de conteúdo, `postagens`:

    ```ts title="src/content/config.ts"
    // Importe utilitários do `astro:content`
    import { z, defineCollection } from "astro:content";

>     // Dina um `type` e `schema` para cada coleção

Reviewer: ```suggestion
    // Defina um `type` e `schema` para cada coleção
```

:::caution
**Não recomendado.** Os plugins Remark e rehype acessam o frontmatter de documentos Markdown ou MDX _cru_. Isso significa que o frontmatter `remarkPluginFrontmatter` é manuseado separadamente do seu `schema` com tipagem correta e não vai refletir quaisquer mudanças ou padrões aplicados pelo Astro. Use por sua conta e risco!
:::

Astro suporta plugins remark ou rehype que [modificam seu frontmatter diretamente](/pt-br/guides/markdown-content/#modificando-frontmatter-programaticamente). Você pode acessar esse frontmatter modificado dentro de uma entrada de conteúdo usando a propriedade `remarkPluginFrontmatter` retornada do `render()`:

```astro "{ remarkPluginFrontmatter }"
---
import { getEntry } from 'astro:content';
const postagemBlog = await getEntry('blog', 'postagem-1');
const { remarkPluginFrontmatter } = await postagemBlog.render();
---
<!-- 
  Esse exemplo assume `tempoLeitura` foi inserido por um plugin remark.
  Veja nosso exemplo "tempo de leitura" para mais detalhes:
  https://docs.astro.build/pt-br/guides/markdown-content/#example-calculate-reading-time
-->
<p>{postagemBlog.data.titulo} — {remarkPluginFrontmatter.tempoLeitura}</p>
```

> Os pipelines do remark e rehype executam quando seu conteúdo é renderizado, que explica o por quê de o `remarkPluginFrontmatter` só estar disponível depois de você chamar `render()` na sua entrada de conteúdo. Em contrate, `getCollection()` e `getEntry()` não pode retornar esses valores diretamente por que eles não renderizam seu conteúdo.

Reviewer: É escrito como "porquê" quando é usado como substantivo.
```suggestion
Os pipelines do remark e rehype executam quando seu conteúdo é renderizado, que explica o porquê de o `remarkPluginFrontmatter` só estar disponível depois de você chamar `render()` na sua entrada de conteúdo. Em contrate, `getCollection()` e `getEntry()` não pode retornar esses valores diretamente porque eles não renderizam seu conteúdo.
```

Author: 😡
